hpi-swt2/lecture-portal
# Feature/#40 starting lectures

This PR introduces the feature of starting and ending lectures

Testing steps:
- as a lecturer, go to "My lectures" (/lectures)
- create a lecture
- click on "Start" in the lectures overview
- students should see the lecture at /lectures/current
- click on "View" in the lectures overview
- click on "End" in the lectures details page
- students should not see the lecture at /lectures/current

Acceptance Criteria:
From ticket #40:
- A lecture that has not been started has the buttons "start" next to it in the "My lectures" list

- Clicking on "Start" next to the lecture in the "My lectures" list will start the associated lecture

- Starting a lecture will add it to the list of currently active lectures in the student view

- After starting the lecture, the lecturer is presented with the dashboard view (tbd) of the newly started lecture

- Once a lecture is running, the lectures list of lectures no longer displays the options "start", but the option "view", next to it, which will take the lecturer directly to the lecture's dashboard

- Nobody else but the lecturer that started the lecture can access the lecturer’s view (dashboard) (tbd)

From ticket #38:
- Clicking on "end lecture" in the lecture view / dashboard ends the lecture

- Ending a lecture removes it from the list of currently active lectures in the student view and removes the option "view" from the lecturers list of lectures

resolves #40, resolves #38 

Steps needed for migration:
- run db:migrate
- PR 64 has to be merged first.
